### PR TITLE
BoolQ

### PR DESCRIPTION
**Pull request by W4ngatang**
Implements BoolQ task. Includes the following: 

- Add BoolQ task (sentence pair classification)
- Add BoolQ prediction writing code

### CODE REVIEW

**Comment by W4ngatang** [comment]
Some minor cleanup (comments, black), but ready for a lookover @sleepinyourhat @pruksmhc

File: main.py [hunk 1]
```diff
@@ -576,6 +576,14 @@ def main(cl_arguments):
                         model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
                     )
                     evaluate_and_write(args, model, [task], splits_to_write)
+            else:
```
Comment (sleepinyourhat) [inline main.py]: @pruksmhc - If this gets merged before your PR, you'll have to make sure these changes don't get incorporated into yours.

File: scripts/superglue-baselines.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ seed=${3:-111}
 gpuid=${2:-0}
 
 function boolq() {
-    python main.py --config config/superglue-bert.conf --overrides "random_seed = ${seed}, cuda = ${gpuid}, run_name = boolq, pretrain_tasks = \"boolq\", target_tasks = \"boolq\", do_pretrain = 0, do_target_task_training = 0, do_full_eval = 1, batch_size = 4, val_interval = 1000"
+    python -m ipdb main.py --config config/superglue-bert.conf --overrides "random_seed = ${seed}, cuda = ${gpuid}, run_name = boolq, pretrain_tasks = \"boolq\", target_tasks = \"boolq\", do_pretrain = 0, do_target_task_training = 0, do_full_eval = 1, batch_size = 4, val_interval = 1000"
```
Comment (sleepinyourhat) [inline scripts/superglue-baselines.sh]: Cut ipdb? :D

File: src/evaluate.py [hunk 3]
```diff
@@ -312,6 +318,25 @@ def _write_winograd_preds(
             preds_fh.write("{0}\n".format(json.dumps(out_d)))
 
 
+def _write_boolq_preds(
```
Comment (sleepinyourhat) [inline src/evaluate.py]: Verify that this is compatible with the eval server.

File: src/tasks/tasks.py [hunk 4]
```diff
@@ -2564,3 +2564,95 @@ def get_metrics(self, reset=False):
...
+        self.test_data_text = None
+
+        self.scorer2 = F1Measure(1)
+        self.scorers = [self.scorer1, self.scorer2]
+        self.val_metric = "%s_acc_f1" % name
+        self.val_metric_decreases = False
+
+    def load_data(self):
+        """ Process the dataset located at path.  """
+        #trg_map = {"true": 1, "false": 0}
```
Comment (sleepinyourhat) [inline src/tasks/tasks.py]: Cut?

File: scripts/superglue-baselines.sh (hunk 2 above)
Comment (W4ngatang) [reply scripts/superglue-baselines.sh]: Oops! Sorry

File: src/tasks/tasks.py [hunk 5]
```diff
@@ -2564,3 +2564,91 @@ def get_metrics(self, reset=False):
...
+                             "passage": passage
+                            }
+                if "answer" in d:
+                    new_datum["answer"] = d["answer"]
+                data.append(new_datum)
+            return data
+
+        self.train_data_text = _load_jsonl(os.path.join(self.path, "train.jsonl"))
+        self.val_data_text = _load_jsonl(os.path.join(self.path, "dev.jsonl"))
+        self.test_data_text = _load_jsonl(os.path.join(self.path, "test.labeled.jsonl"))
```
Comment (W4ngatang) [inline src/tasks/tasks.py]: this should be unlabeled

**Comment by W4ngatang** [comment]
Waiting on confirmation that this works with SuperGLUE eval server, then will merge in.

File: src/evaluate.py (hunk 3 above)
Comment (W4ngatang) [reply src/evaluate.py]: Confirmed!

**Pull request merged**